### PR TITLE
OBA-403: Add liquibase migration schemas

### PR DESCRIPTION
**Pull request by elroman**

### CODE REVIEW

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/RequestInfoEntity.java [hunk 1]
```diff
@@ -2,22 +2,26 @@
...
+import javax.persistence.*;
 import java.time.LocalDate;
 
 @Data
 @Entity
 public class RequestInfoEntity {
     @Id
-    @GeneratedValue(strategy = GenerationType.SEQUENCE)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "request_info_generator")
+    @SequenceGenerator(name = "request_info_generator", sequenceName = "request_info_id_seq", allocationSize = 1)
```
Comment (valb3r) [inline fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/RequestInfoEntity.java]: Using allocationSize 1 is a bad idea from a performance perspective as it is an entity-per-request class. Use 50 as sane default. See 
https://stackoverflow.com/questions/54741182/what-does-the-jpa-sequencegenerator-allocationsize-mean
Probably you will need to add `INCREMENTED BY` to liquibase sequence definition

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/RedirectUrlsEntity.java [hunk 2]
```diff
@@ -2,20 +2,25 @@
...
-import javax.persistence.Id;
+import javax.persistence.*;
 
 @Data
 @Entity
 public class RedirectUrlsEntity {
     @Id
-    @GeneratedValue(strategy = GenerationType.SEQUENCE)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "redirect_urls_generator")
+    @SequenceGenerator(name = "redirect_urls_generator", sequenceName = "redirect_urls_id_seq", allocationSize = 1)
```
Comment (valb3r) [inline fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/RedirectUrlsEntity.java]: Using allocationSize 1 is a bad idea from a performance perspective as it is an entity-per-request class. Use 50 as sane default. See below

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/LoginEntity.java [hunk 3]
```diff
@@ -18,8 +15,10 @@
 @Builder
 public class LoginEntity {
     @Id
-    @GeneratedValue(strategy = GenerationType.SEQUENCE)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "login_generator")
+    @SequenceGenerator(name = "login_generator", sequenceName = "login_id_seq", allocationSize = 1)
```
Comment (valb3r) [inline fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/LoginEntity.java]: Using allocationSize 1 is a bad idea from a performance perspective as it is an entity-per-request class. Use 50 as sane default. See below

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/CookieEntity.java [hunk 4]
```diff
@@ -21,9 +18,13 @@
 @ToString
 public class CookieEntity {
     @Id
-    @GeneratedValue(strategy = GenerationType.SEQUENCE)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "cookie_generator")
+    @SequenceGenerator(name = "cookie_generator", sequenceName = "cookie_id_seq", allocationSize = 1)
```
Comment (valb3r) [inline fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/CookieEntity.java]: Using allocationSize 1 is a bad idea from a performance perspective as it is an entity-per-request class. Use 50 as sane default. See below

File: fintech-examples/fintech-db-schema/util/pg-create-schema.sh [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
```
Comment (valb3r) [inline fintech-examples/fintech-db-schema/util/pg-create-schema.sh]: This script requires local pgsql installation. Why not to initialize postres using docker init?
i.e.:
docker run ... open_banking?TC_DAEMON=true&TC_INITSCRIPT=init.sql

File: fintech-examples/fintech-db-schema/liquibase.example.properties [hunk 6]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (valb3r) [inline fintech-examples/fintech-db-schema/liquibase.example.properties]: Do we need this?

File: fintech-examples/fintech-db-schema/README.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
+# How to work with liquibase:
+
+* If you migrate to liquibase from our project versions that are earlier than 1.4, delete all tables and sequences in your postgres database (if you already have them). Better do it with such tools as pgAdmin or Dbeaver to be sure that you deleted everything (or use command line).
+
+* If you migrate to liquibase from our project versions that are earlier than 1.12, delete all tables and sequences in your postgres database (if you already have them). Better do it with such tools as pgAdmin or Dbeaver to be sure that you deleted everything (or use command line).
+
+* Create new scheme 'fintech' in DB
```
Comment (valb3r) [inline fintech-examples/fintech-db-schema/README.md]: Suggest to remove paragraphs above

File: fintech-examples/fintech-db-schema/README.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
+# How to work with liquibase:
```
Comment (valb3r) [inline fintech-examples/fintech-db-schema/README.md]: When using liquibase you should do hibernate validation:
```
jpa:
    hibernate:
      naming:
        physical-strategy: de.adorsys.opba.db.hibernate.PrefixAndSnakeCasePhysicalNamingStrategy
      # DDL is handled by Liquibase
      ddl-auto: validate # <<< THIS!
```

File: fintech-examples/fintech-server/src/main/resources/application.yml [hunk 9]
```diff
@@ -19,10 +19,11 @@ mock:
...
   datasource:
-    url: jdbc:h2:mem:test
-    username: SA
-    password:
-    driver-class-name: org.h2.Driver
+    url: jdbc:postgresql://localhost/open_banking?currentSchema=fintech
+    username: postgres
+    password: docker
+  liquibase:
+    change-log: classpath:master.xml
```
Comment (valb3r) [inline fintech-examples/fintech-server/src/main/resources/application.yml]: Add `enabled: true`. No one on DEV will perform manual migrations. Liquibase should be disabled only on PROD (and even there it is questionable decision)

File: fintech-examples/fintech-db-schema/pom.xml [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+                <groupId>org.liquibase</groupId>
+                <artifactId>liquibase-maven-plugin</artifactId>
+                <version>${liquibase.version}</version>
+                <configuration>
+                    <propertyFile>${propertyFile}</propertyFile>
+                </configuration>
+
+                <dependencies>
+                    <dependency>
+                        <groupId>org.postgresql</groupId>
```
Comment (valb3r) [inline fintech-examples/fintech-db-schema/pom.xml]: Doesn't seem to be needed in this module

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/RequestInfoEntity.java (hunk 1 above)
Comment (elroman) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/RequestInfoEntity.java]: done

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/RedirectUrlsEntity.java (hunk 2 above)
Comment (elroman) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/RedirectUrlsEntity.java]: done

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/LoginEntity.java (hunk 3 above)
Comment (elroman) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/LoginEntity.java]: done

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/CookieEntity.java (hunk 4 above)
Comment (elroman) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/CookieEntity.java]: done

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/hibernate/PrefixAndSnakeCasePhysicalNamingStrategy.java [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+import com.google.common.base.CaseFormat;
+import org.hibernate.boot.model.naming.Identifier;
+import org.hibernate.boot.model.naming.PhysicalNamingStrategyStandardImpl;
+import org.hibernate.engine.jdbc.env.spi.JdbcEnvironment;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Converts Hibernate snakeCase naming into Database-friendly camel_case, also adds prefix to the tables.
```
Comment (valb3r) [inline fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/hibernate/PrefixAndSnakeCasePhysicalNamingStrategy.java]: Change to
`Converts Hibernate camelCase naming into Database-friendly snake_case, also adds prefix to the tables.`

**Pull request merged**